src/components/ebay-area-chart/component.js

```diff
@@ -0,0 +1,331 @@
+import Highcharts from 'highcharts';
+import accessibility from 'highcharts/modules/accessibility';
+import patternFill from 'highcharts/modules/pattern-fill';
+import {
+    backgroundColor,
+    gridColor,
+    labelsColor,
+    legendColor,
+    legendInactiveColor,
+    legendHoverColor,
+    tooltipBackgroundColor,
+    tooltipShadows,
+    setSeriesColors,
+} from '../../common/charts/colors';
+
+import { ebayLegend } from '../../common/charts/legend';
+
+if (typeof Highcharts === 'object') {
+    // enable highcharts accessibility with wrapper function
+    accessibility(Highcharts);
+    // patternFill highcharts wrapper function enables rendering patterns instead of just solid colors
+    patternFill(Highcharts);
+    // add custom legend wrapper function
+    ebayLegend(Highcharts);
+}
+const pointSize = 1.5;
+
+export default class {
+    onCreate() {
+        this.chartRef = null;
+        this.state = {
```

Comment:
Do not use state unless we change it later. Either set it as a getter or use a common component method.

Reply:
this state variable is used in the marko template, should I still move it?  I wanted easy access to the variable from the marko markup

Reply:
state is only used when we want to rerender based on it's changes.
If we want to access something from a template, we can put it in a method on the `component`.
So like:
```
getContainerId() {
   return `ebay-bar-chart-${this.id}`;
}
```
And then in the template we call it using:
`<div id=component.getContainerId()>`